src/theory/quantifiers/sygus/synth_verify.cpp

```diff
@@ -111,12 +111,29 @@ Result SynthVerify::verify(Node query,
         Trace("cegqi-debug") << "...squery : " << squery << std::endl;
         squery = d_tds->rewriteNode(squery);
         Trace("cegqi-debug") << "...rewrites to : " << squery << std::endl;
-        // if the query simplifies to false, then something
-        // went wrong or the answer is unknown (for debugging).
-        if (squery.isConst() && !squery.getConst<bool>())
+        if (!squery.isConst() || !squery.getConst<bool>())
         {
-          Assert(r.getStatus() == Result::UNKNOWN)
-              << "Expected model from verification step to satisfy query";
+          if (squery.isConst())
+          {
+            // simplified to false, the result should have been unknown, or
+            // else this indicates a check-model failure.
+            Assert(r.getStatus() == Result::UNKNOWN)
+                << "Expected model from verification step to satisfy query";
+          }
+          // If the query did not simplify to true, then it may be that the
+          // value for an oracle function was not what we expected.
+          if (options().quantifiers.oracles)
+          {
+            // In this case, we reconstruct the query, which may include more
+            // information about oracles than we had previously. We rerun the
+            // satisfiability check above.
```

Comment:
Clarify that this is (if I understand correctly) due to the I/O pairs found during search being added to the query.

Reply:
The I/O pairs are actually added in `TermDbSygus::rewriteNode`.  I've commented.